source.go

```diff
@@ -232,12 +234,33 @@ func (s *PrefixSource) Get(ctx context.Context, path ...string) (interface{}, bo
 type MultiSource []Source
 
 // Get a value from the ordered set of Sources.
-func (s MultiSource) Get(ctx context.Context, path ...string) (interface{}, bool) {
-	for _, ss := range s {
-		v, found := ss.Get(ctx, path...)
+func (ms MultiSource) Get(ctx context.Context, path ...string) (interface{}, bool) {
+	var p, _ = regexp.Compile(envPattern)
+	var v interface{}
+	var found bool
+	for _, ss := range ms {
+		v, found = ss.Get(ctx, path...)
 		if found {
-			return v, found
+			break
 		}
 	}
-	return nil, false
+
+	if s, ok := v.(string); ok && p.Match([]byte(s)) {
+		key := ms.unwrap([]byte(s))
+		return ms.Get(ctx, strings.Split(string(key), "_")...)
+	}
+
+	return v, found
+}
+
+const (
+	envPattern = `\${[^}]+}`
+)
+
+func (ms MultiSource) unwrap(source []byte) []byte {
+	var p, _ = regexp.Compile(envPattern)
+	return p.ReplaceAllFunc(source, func(match []byte) []byte {
+		name := match[2 : len(match)-1] // strip ${}
+		return []byte(string(name))
```

Comment:
Is the `[]byte` casting needed here? Are match and name already `[]byte` types?

Reply:
oh.  Good eyes.  Yeah there's like double casting there.  I'll remove it.